t/t1029-mf-priority-default-bank.t

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+test_description='test adding default bank to jobspec in multi-factor priority plugin'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SUBMIT_AS=${SHARNESS_TEST_SRCDIR}/scripts/submit_as.py
+DB_PATH=$(pwd)/FluxAccountingTest.db
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+	flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'check that mf_priority plugin is loaded' '
+	flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'create flux-accounting DB' '
+	flux account -p $(pwd)/FluxAccountingTest.db create-db
+'
+
+test_expect_success 'start flux-accounting service' '
+	flux account-service -p ${DB_PATH} -t
+'
+
+test_expect_success 'add some banks to the DB' '
+	flux account -p ${DB_PATH} add-bank root 1 &&
+	flux account -p ${DB_PATH} add-bank --parent-bank=root account1 1 &&
+	flux account -p ${DB_PATH} add-bank --parent-bank=root account2 1
+'
+
+test_expect_success 'add a user to the DB' '
+	flux account -p ${DB_PATH} add-user --username=user1001 --userid=1001 --bank=account1 &&
+	flux account -p ${DB_PATH} add-user --username=user1001 --userid=1001 --bank=account2
+'
+
+test_expect_success 'submit a job before plugin has any flux-accounting information' '
+	jobid=$(flux python ${SUBMIT_AS} 1002 hostname) &&
+	flux job info $jobid eventlog > eventlog.out &&
```

Comment:
I think there is a race condition here because the job may not have reached the DEPEND state by the time `flux job info` is run. Maybe `flux job wait-event -vt 60 $jobid depend` would be better here? (Sorry if I originally had suggested `flux job info`!)

I'm not actually sure how to definitively test that a job _stays_ in DEPEND state though.. so maybe right after the wait-event you can dump the eventlog and check again like you have here now?

Reply:
Thanks for this suggestion! Sorry that I was using the wrong method to check for the job state here. I just re-ran this set of tests (both under and not under valgrind) and using `flux job wait-event` seemed to work. I can then check the eventlog afterwards and look for the "depend" event. I've just pushed up this change. Let me know if it looks better!

On a side note, I should probably step through the test suite tomorrow (like you mentioned in #294) and change the tests that test for job states to use the suggestion above. I'll open a separate issue on that.